### PR TITLE
[RFC] Add information about guicursor

### PR DESCRIPTION
**Pull request by tjdevries**
Add some information to `:CheckHealth` about the changes in guicursor.

Might not be required, but we can just tell people now to run `:CheckHealth` and check the suggestions.

### CODE REVIEW

**Comment by tjdevries** [comment]
Also doesn't show up if they don't have it set, so it should be pretty non-spammy for people who have already read `Following-Head`.

File: runtime/autoload/health/nvim.vim [hunk 1]
```diff
@@ -159,10 +159,28 @@ function! s:check_terminal() abort
...
 endfunction
 
+function! s:check_guicursor() abort
+  let environment_variable = $NVIM_TUI_ENABLE_CURSOR_SHAPE
+
+  if environment_variable != ""
+    call health#report_start('guicursor')
+    call health#report_info('If you are experiencing issues with cursor shape, check the following information:')
+    call health#report_info('Current $NVIM_TUI_ENABLE_CURSOR_SHAPE value is: ' . environment_variable)
+    call health#report_info('Current term,shell is: ' . $TERM . ', ' . &shell)
```
Comment (justinmk) [inline runtime/autoload/health/nvim.vim]: Already part of check_terminal

File: runtime/autoload/health/nvim.vim [hunk 2]
```diff
@@ -159,10 +159,28 @@ function! s:check_terminal() abort
...
+  endif
+endfunction
+
 function! health#nvim#check() abort
   call s:check_config()
   call s:check_performance()
   call s:check_rplugin_manifest()
   call s:check_terminal()
   call s:check_tmux()
+  call s:check_guicursor()
```
Comment (justinmk) [inline runtime/autoload/health/nvim.vim]: Should be called from s:check_config

File: runtime/autoload/health/nvim.vim [hunk 3]
```diff
@@ -159,10 +159,28 @@ function! s:check_terminal() abort
   endif
 endfunction
 
+function! s:check_guicursor() abort
+  let environment_variable = $NVIM_TUI_ENABLE_CURSOR_SHAPE
+
+  if environment_variable != ""
+    call health#report_start('guicursor')
```
Comment (justinmk) [inline runtime/autoload/health/nvim.vim]: Don't need to start a new report, should be in configuration report

**Comment by tjdevries** [comment]
@justinmk  Addressed the comments (is there a button to press that can signal that?)

**Comment by justinmk** [comment]
Merged

**Pull request closed without merging**